_posts/2019-10-29-dhis2-and-docker.md

```diff
@@ -0,0 +1,214 @@
+
+To destroy the instance, run `docker-compose down`. 
+### DHIS2 with pre-populated postgres database using docker-compose
+
+To achieve this, you will need an SQL file with the schema and data you want to pre-populate postgres with. You will attach that file to postgres container as a volume. Here's an example of the `docker-compose.yml` file. 
+
+    version: '3'  
+	  services:  
+	    db:  
+	      image: mdillon/postgis:10-alpine  
+		  command: postgres -c max_locks_per_transaction=100  
+		  environment:  
+	        POSTGRES_USER: dhis  
+            POSTGRES_DB: dhis2  
+            POSTGRES_PASSWORD: dhis
+          volumes:
+            - ./config/init.sql:/docker-entrypoint-initdb.d/init.sql
+        web:  
+          image: dhis2/core:2.33.0  
+		  volumes:  
+		    - ./config/dhis2_home/dhis.conf:/DHIS2_home/dhis.conf
+		  environment:
+		    - WAIT_FOR_DB_CONTAINER=db:5432 -t 0
+	      ports:  
+		    - "8080:8080"
+		  depends_on: 
+		    - db
+
+You probably noticed, that most of the file is the same as in the previous step, but we added _volumes_ and _environment_ properties for _db_ and _web_ containers. 
+
+Attaching the SQL file to the _db_ container will tell postgres to pre-populate db before starting the container. Depending on the file size, postgres might take a while to do this and we might run into race conditions when DHIS2 starts earlier than postgres and fails to connect. To solve this, we tell the _web_  container that it has to wait for postgres to finish before starting. We do this by using the WAIT_FOR_DB_CONTAINER environment variable.  
+You will notice the -t  option. The -t option stands for  _timeout_. 0  means that _web_ container will wait as long as it takes for _db_ container to become ready, but you can set any timeout (in seconds) you want.  
+
+Note, that you still need to configure connection properties in dhis.conf: 
+
+    connection.dialect = org.hibernate.dialect.PostgreSQLDialect  
+	connection.driver_class = org.postgresql.Driver  
+	
+	# "db" maps to service name defined in docker-compose
+	# "dhis2" maps to POSTGRES_DB defined in docker-compose
+	connection.url = jdbc:postgresql://db/dhis2 
+	
+	# maps to POSTGRES_USER environment variable in docker-compose.
+	connection.username = dhis 
+	
+	# maps to POSTGRES_PASSWORD environment variable in docker-compose.
+	connection.password = dhis 
+
+
+To run this file with docker compose, execute the following command: 
+`docker-compose -f /pathToDockerComposeFile up`
+
+or, if the file is on your current directory: 
+`docker-compose up`
+
+**Outcome**: docker-compose will start 2 containers: _db_ and _web_ and when tomcat fully starts up, you will be able to reach DHIS2 at http://localhost:8080. The database will be pre-populated. 
+
+To destroy the instance, run `docker-compose down`. 
```

Comment:
Would be nice to have a line about how to remove instances and cleaning up when tearing down a cluster.

Reply:
I think that it would be totally random, since I'm not talking about the cluster in this guide. 
Feel free to update with more cases using the cluster. I rarely use it, so don't feel I can write a guide about it.

Reply:
Alright. :+1: